benchmark/sparse/linear_regression/inference_benchmark.md

```diff
@@ -0,0 +1,91 @@
+| :--------------- | :----------------------------------------------------------- |
+| Keras            | v2.2.4                                                      |
+| TensorFlow       | v1.11.0                                                     |
+| MXNet-mkl         | v1.3.0   
+
+#### CPU
+##### Speed
+| Instance Type | GPUs  | Batch Size  | Keras-MXNet Time/Batch) | Keras-TensorFlow Time/Batch)  |
+|-----|-----|-----|-----|-----|
+| C5.8XLarge |   0  | 64  | 64.2 sec | 64.3 sec
+| C5.8XLarge |   0  | 128 | 32.1 sec | 31.4 sec 
+| C5.8XLarge |   0  | 256 | 16.5 sec | 16.3 sec
+| C5.8XLarge |   0  | 512 | 8.3 sec | 8.8 sec 
+| C5.8XLarge |   0  | 1024 | 4.3 sec | 4.4 sec
+
+#### Memory utilization
+| Instance Type | GPUs  | Batch Size | Keras-MXNet (Mem %) | Keras-TensorFlow (Mem %)  |
+|-----|-----|-----|-----|-----|
+| C5.8XLarge |   0  | 64  | 0.4 | 0.5 |
+| C5.8XLarge |   0  | 128 | 0.4 | 0.5 | 
+| C5.8XLarge |   0  | 256 | 0.4 | 0.5  |
+| C5.8XLarge |   0  | 512 | 0.4 | 0.5 |
+| C5.8XLarge |   0  | 1024 | 0.4 | 0.5 |
+
+#### Memory consumed
+| Instance Type | GPUs  | Batch Size | Keras-MXNet (MB) | Keras-TensorFlow (MB)  |
+|-----|-----|-----|-----|-----|
+| C5.8XLarge |   0  | 64  | 1630.8 | 1573.8 |
+| C5.8XLarge |   0  | 128 | 1574.7 | 1561.2 | 
+| C5.8XLarge |   0  | 256 | 1477.8 | 1501.4  |
+| C5.8XLarge |   0  | 512 | 1407.0| 1472.5 |
+| C5.8XLarge |   0  | 1024 | 1336.3 | 1466.8 |
+
+#### GPU
+### Configuration
+##### Input samples are sparse feature vectors with 10,000 dimensions
+| Dataset          | Synthetic(Randomly generated)                                |
+| :--------------- | :----------------------------------------------------------- |
+| Keras            | v2.2.4                                                      |
+| TensorFlow-GPU   | v1.11.0                                                     |
+| MXNet-cu90mkl    | v1.3.0                                                      |
+
+##### Single GPU
+##### Speed
+| Instance Type | GPUs  | Batch Size  | Keras-MXNet (Time/Batch) | Keras-TensorFlow Time/Batch)  |
+|-----|-----|-----|-----|-----|
+| P3.8XLarge |   1  | 64  | 3.0 sec | 2.0 sec
+| P3.8XLarge |   1  | 128 | 2.3 sec | 1.2 sec 
+| P3.8XLarge |   1  | 256 | 1.2 sec | 0.7 sec
+| P3.8XLarge |   1  | 512 | 0.8 sec | 0.5 sec
+| P3.8XLarge |   1  | 1024 | 0.4 sec | 0.4 sec
+
+##### Memory utilization
+| Instance Type | GPUs  | Batch Size | Keras-MXNet (GPU Utilization %) | Keras-TensorFlow (GPU Utilization %)  |
+|-----|-----|-----|-----|-----|
+| P3.8XLarge |   1  | 64  | 9 | 7
+| P3.8XLarge |   1  | 128 | 8 | 7
+| P3.8XLarge |   1  | 256 | 8 | 7
+| P3.8XLarge |   1  | 512 | 8 | 7
+| P3.8XLarge |   1  | 1024 | 7 | 8
+
+##### Memory consumed
+| Instance Type | GPUs  | Batch Size | Keras-MXNet (GPU Memory (MB)) | Keras-TensorFlow (GPU Memory (MB))  |
+|-----|-----|-----|-----|-----|
+| P3.8XLarge |   1  | 64  | 966.7 | 3322.9
+| P3.8XLarge |   1  | 128 | 970.9 | 3053.4
```

Comment:
I observe TF backend always take all available GPU memories by default. around 15388MiB for each GPU. This is expected no matter what batch size you use. Please double check, especially double check `nvidia-smi` is used during the middle of inference and training. You will get different numbers during data prep and evaluation (model.evaluate(xxx))

Reply:
Yes updated the numbers